inventory/any.cf

```diff
@@ -524,34 +524,35 @@ bundle agent cfe_autorun_inventory_aws_ec2_metadata
       "cfe_aws_ec2_metadata_from_cache";
 }
 bundle agent cfe_autorun_inventory_aws_ec2_metadata_cache
-# @brief Cache ec2 metadata from http request for one day
+# @brief Cache ec2 metadata from http request
+# It's up to caller to ensure this bundle is called only once a day
 #
 # Provides cache of ec2 instance metadata for inventory
 {
 
   vars:
-
     ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
-
-      "curl"      string => "$(paths.curl)";
-      "curl_opts" string => "--max-time 1 --connect-timeout 1 --silent";
       "URL"       string => "http://169.254.169.254/latest/dynamic/instance-identity/document";
       "cache"     string => "$(sys.statedir)/aws_ec2_metadata";
+      "response"
+        data => url_get($(URL), '{"url.max_content": 512000, "url.timeout": 1}'),
+	# To prevent `url_get` from firing on every agent run, this variable
+	# depends on a dummy command which does nothing but fires only once a day
+	depends_on => { "daily_dummy_job" };
 
   commands:
-
-    ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
-
-      "$(curl)"
-        args => "$(curl_opts) $(URL) -o $(cache)",
-        if => and( not( fileexists( $(cache) )),
-                   isexecutable( $(curl) ));
+    "/bin/true"
+	# dummy command which does nothing but fires only once a day
```

Comment:
This will be recorded as a /repair/. I think this one should be interpreted as KEPT instead.

```
      classes => kept_successful_command,
```